### DISCUSSION TITLE
List (or count) containers on a specific worker

### DISCUSSION THREAD

**Discussion by wanderanimrod** [concourse/concourse#8926] [General]
## Question
Hello, is there a way to list containers on a specific worker **_(regardless of team)_** either via `fly` or ATC api? 

## Context
We are running into the `max containers reached` error a lot. We have set the `maxContainers` limit on our workers to a high number (700) and set the container placement strategy to `limitActiveContainers` on the web nodes to a much lower number (340), but we still run into the `max containers reached` error.

I wanted to know how many containers are running for sure on a worker, but I don't see any way in the API to do that. There is a fly command for listing containers for a team, but none for a worker.

### Calling the ATC api via fly
```bash
fly -t main curl '/api/v1/containers/report?worker_name=concourse-worker-2' -- -v -XPUT -H "Content-type:application/json"

#and

fly -t main curl /api/v1/containers/report -- -v -XPUT -d '{"worker_name": "concourse-worker-15"}' -H "Content-type:application/json"
```
Both of these returns a 404, but this [line of code](https://github.com/concourse/concourse/blob/master/atc/routes.go#L226C10-L226C35) appears to suggest that the second command should work, and there's [a very old test](https://github.com/concourse/concourse/blob/a0f72b6995194b595023efc15dd608c4553c0559/tsa/cmd/tsa/report_containers_test.go#L33) that appears to suggest that the first one should work.

## Suspicion
I am starting to suspect that the apparent failure of the web nodes to limit active containers might be related to this api not working.

## Ask
Is there a way to list containers on a specific worker **_(regardless of team)_** either via `fly` or ATC api?

_Note: I know that when you run `fly workers`, you get the number of containers on the worker in the table. I can't use this method to track the containers metric. I need a way to get containers for a specific worker. The `workers_containers` prometheus metric also seems off._

**Comment by taylorsilva** [comment]
`fly workers` is what you want:

```
┌─[taylor-macbook][~/workspace]
└─▪ fly -t ci workers
name                                  containers  platform  tags        team  state    version  age
855c7ffd-a80e-4224-90fa-e4bc74e26ed4  1           linux     bosh        main  running  2.5      265d
administrators-Mac-mini.local         0           darwin    none        main  running  2.5      21d
ci-pr-worker-0                        13          linux     pr          none  running  2.5      12d
ci-workers-worker-0                   79          linux     none        none  running  2.5      11d
ci-workers-worker-1                   76          linux     none        none  running  2.5      11d
ci-workers-worker-2                   71          linux     none        none  running  2.5      11d
ci-workers-worker-3                   69          linux     none        none  running  2.5      11d
ci-workers-worker-4                   83          linux     none        none  running  2.5      11d
topgun-worker-worker-0                0           linux     k8s-topgun  none  running  2.5      12d
windows-worker-ci                     2           windows   none        main  running  2.5      13d
```

Add the `--json` flag if you want to do further parsing. Also includes more info like active volumes and tasks.

That said, if you're running into max container errors, you've likely out grown the number of workers you've currently deployed. The only sure way to know will be to track metrics.
